examples/mauna-loa/script.jl

```diff
@@ -0,0 +1,302 @@
+#     therefore call `f(xtrain)` which implies zero (additional) observation
+#     noise.
+#
+#     Alternatively, we could have omitted the `θ.noise_scale^2 *
+#     WhiteKernel()` term and instead passed the noise variance as a second
+#     argument to the GP call in `build_finite_gp`, `f(xtrain,
+#     θ.noise_scale^2)`.
+#
+#     These two approaches have slightly different semantics: In the first one,
+#     the `WhiteKernel` contributes non-zero variance to the `[i, j]` element
+#     of the covariance matrix of the `FiniteGP` if `xtrain[i] == xtrain[j]`
+#     (based on the values of the features). In the second one, the observation
+#     noise variance passed to `FiniteGP` only contributes to the diagonal
+#     elements of the covariance matrix, i.e. for `i == j`.
+#
+#     Moreover, the variance (uncertainty) of the posterior predictions
+#     includes the variance from the `WhiteKernel`, but does not include the
+#     variance of the observation noise passed to the `FiniteGP`. To include
+#     the observation noise in posterior predictions from the second approach,
+#     call `fpost_opt(xtest, noise_scale^2)`.
+#
+# !!! tip
+#     In practice, we recommend that you pass in observation noise to the
+#     `FiniteGP`, and omit the explicit `WhiteKernel`.
+
+# We obtain the posterior, conditioned on the (finite) observations, by calling
+# [`posterior`](@ref):
+
+function build_posterior_gp(θ)
+    fx = build_finite_gp(θ)
+    return posterior(fx, ytrain)
+end
+#md nothing #hide
+
+# Now we can put it all together to obtain a [`PosteriorGP`](@ref).
+# The call to `ParameterHandling.value` is required to replace the constraints
+# (such as `positive` in our case) with concrete numbers:
+
+fpost_init = build_posterior_gp(ParameterHandling.value(θ_init))
+
+# Let's visualize what the GP fitted to the data looks like, for the initial choice of kernel hyperparameters.
+#
+# We use the following function to plot a GP `f` on a specific range, using the
+# AbstractGPs [plotting
+# recipes](https://juliagaussianprocesses.github.io/AbstractGPs.jl/dev/concrete_features/#Plotting).
+# By setting `ribbon_scale=2` we visualize the uncertainty band with :math:`\pm
+# 2` (instead of the default :math:`\pm 1`) standard deviations.
+
+plot_gp!(f; label) = plot!(f(1920:0.2:2030); ribbon_scale=2, linewidth=1, label)
+#md nothing #hide
+
+# !!! tip
+#     The `let` block [creates a new
+#     scope](https://docs.julialang.org/en/v1/manual/variables-and-scoping/#scope-of-variables),
+#     so any utility variables we define in here won't leak outside. This is
+#     particularly helpful to keep notebooks tidy! The return value of this
+#     block is given by its last expression.
+
+let
+    plotdata()
+    label = "posterior f(⋅)"
+    plot_gp!(fpost_init; label)  # this returns the current plot object
+end  # and so the plot object will be shown
+## `label` is not available outside the `let` scope
```

Comment:
:smile: You just added this artificial variable so the comment is somewhat more reasonable, didn't you? I really don't think this makes sense at all. `let` can be useful for the reasons described in the comment but there's no reason for a block here - I think it's even worse with this artificial variable since I don't think anyone would write code in this way.

It is also a bit unfortunate to use a syntax here (`; label)`) that only works with more recent versions of Julia (IIRC Julia >= 1.4).

Please just replace this with the following lines:
```suggestion
plotdata()
plot_gp!(fpost_init; label="posterior f(⋅)")
```